File: test/cpp/test_2709.cpp
#include "../../src/cpp/code_2709.cpp"
#include "cpp_deps/boilerplate.h"

void test(Solution& sol, const json& input, const json& output) {
    vector<int> nums = input["nums"].get<vector<int>>();
    bool expected = output.get<bool>();
    bool result = sol.canTraverseAllPairs(nums);
    CHECK_EQ(result, expected);
}

TEST_CASE("") {
    TEST("test/test_json/test_2709.json");
}

File: test/python/test_2709.py
import sys
import os

sys.path.append(os.getcwd())
from .python_deps.boilerplate import BoilerTest
from src.python.code_2709 import Solution


def input_fetcher(case):
    return case["input"]["nums"]


def test():
    test = BoilerTest(2709, input_fetcher)
    solution = Solution().canTraverseAllPairs
    test.run_tests(solution)
